### PR TITLE
Code review changes

### PR DESCRIPTION
**Pull request by mohansisodiya50**

### CODE REVIEW

File: dev/index.jsx [hunk 1]
```diff
@@ -1,108 +1,44 @@
 import React, { Component } from 'react';
 import ReactDOM from "react-dom";
 import _ from 'lodash';
-import Autosuggest from 'react-autosuggest';
 import MovieDetails from './movieDetails.jsx';
+import MovieSuggestions from './movieSuggestions.jsx';
 
-class Search extends Component {
+class Index extends Component {
```
Comment (newpatriks) [inline dev/index.jsx]: you shouldn't have changed this name. When you are loading this on `app.js` you name it Search, so you can leave it with its name. Index it's not a good name.

File: dev/movieDetails.jsx [hunk 2]
```diff
@@ -2,13 +2,38 @@ import React, { Component } from 'react';
...
-class MovieDetails extends Component{
-	componentWillMount() {
-	  document.addEventListener('click', this.handleClick, false);
+class MovieDetails extends Component {
+
+	constructor(props) {
+		super(props);
+		this.state = { data:''};
+		this.getmovieDetails = this.getmovieDetails.bind(this);
+		this.getmovieDetails(278927);
```
Comment (newpatriks) [inline dev/movieDetails.jsx]: This shouldn't be here. If what you want is to load a default movie you need to use the parameter you're sending from the parent component to this component. So in the `Search` constructor you need to set the state to the default value. I added a comment there too.

File: dev/index.jsx [hunk 3]
```diff
@@ -1,108 +1,44 @@
...
-          revenue: responseJson.revenue
-        });
-      });
-  this.setState({ showResults: false });
-  }
-
-  componentDidMount() {
-    this.getmovieDetails(278927);
+    this.state = {
+      movieId: '',
```
Comment (newpatriks) [inline dev/index.jsx]: This should be set to a default value. You are doing this on the movieDetails component constructor right now (and it's wrong). Just do this here: 
```
this.state = {
  movieId: '278927',
  ...
}
```

File: dev/index.jsx (hunk 1 above)
Comment (mohansisodiya50) [reply dev/index.jsx]: done.

File: dev/index.jsx [hunk 4]
```diff
@@ -1,108 +1,46 @@
...
     return (
       <div>
         <div  className="searchBar">
-          <input className="movieInput" placeholder="Search Movie..."  value={this.state.term} id={this.state.id} onInput={event => this.onInputChange(event.target.value)} />
-          {self.state.showResults ? <ul className="suggestionDropDown">{movieList}</ul> : "" }
+          <input className="movieInput" placeholder="Search Movie..."  value={this.state.movieTitle} onInput={event => this.onInputChange(event.target.value)} />
         </div>
-          <MovieDetails data={this.state}/>
+        <MovieSuggestions suggestionsVisisble={this.state.suggestionsVisisble} suggestionSelected={this.handleSuggestionSelected} movieTitle={ this.state.movieTitle }/>
+        <MovieDetails getDetails={ this.state.getDetails }movieId={ this.state.movieId }/>
```
Comment (newpatriks) [inline dev/index.jsx]: space

**Pull request merged**